lib/plugins/entry-files-manifest.js

```diff
@@ -13,32 +13,66 @@ const PluginPriorities = require('./plugin-priorities');
-    // assets (which will have their original structure). We
-    // delete the entrypoints key, and then process the new assets
-    // like normal below
-    delete assets.entrypoints;
-
-    // This will iterate over all the entry points and convert the
-    // one file entries into an array of one entry since that was how the entry point file was before this change.
-    for (const asset in assets) {
-        for (const fileType in assets[asset]) {
-            if (!Array.isArray(assets[asset][fileType])) {
-                assets[asset][fileType] = [assets[asset][fileType]];
+function processOutput(webpackConfig) {
+    return (assets) => {
+        for (const entry of [copyEntryTmpName, sharedEntryTmpName]) {
+            delete assets[entry];
+        }
+
+        // with --watch or dev-server, subsequent calls will include
+        // the original assets (so, assets.entrypoints) + the new
+        // assets (which will have their original structure). We
+        // delete the entrypoints key, and then process the new assets
+        // like normal below
+        delete assets.entrypoints;
+
+        // This will iterate over all the entry points and convert the
+        // one file entries into an array of one entry since that was how the entry point file was before this change.
+        const integrity = {};
+        const integrityAlgorithm = webpackConfig.integrityAlgorithm;
+        const publicPath = webpackConfig.getRealPublicPath();
+
+        if (integrityAlgorithm) {
+            integrity.algorithm = integrityAlgorithm;
+            integrity.hashes = {};
+        }
+
+        for (const asset in assets) {
+            for (const fileType in assets[asset]) {
+                if (!Array.isArray(assets[asset][fileType])) {
+                    assets[asset][fileType] = [assets[asset][fileType]];
+                }
+
+                if (webpackConfig.integrityAlgorithm) {
+                    for (const file of assets[asset][fileType]) {
+                        if (!(file in integrity.hashes)) {
+                            const filePath = path.resolve(
+                                webpackConfig.outputPath,
+                                file.replace(publicPath, '')
+                            );
+
+                            if (fs.existsSync(filePath)) {
+                                const hash = crypto.createHash(webpackConfig.integrityAlgorithm);
+                                const fileContent = fs.readFileSync(filePath, 'utf8');
+                                hash.update(fileContent, 'utf8');
+
+                                integrity.hashes[file] = hash.digest('base64');
+                            }
+                        }
+                    }
+                }
             }
         }
-    }
 
-    return JSON.stringify({
-        entrypoints: assets
-    }, null, 2);
+        return JSON.stringify({
+            entrypoints: assets,
+            integrity
```

Comment:
Should we include the `integrity` key at all if the feature is disabled?

Reply:
The idea was to be able to detect if Encore supported that feature to make the bundle display a relevant error if someone tried to use it with an old version of Encore.

But since I also agree about not making it optional in the bundle anymore (by detecting if `integrity` is present in the manifest) that's not needed anymore :)